egs/wsj/s5/steps/nnet3/get_egs_discriminative.sh

```diff
@@ -0,0 +1,467 @@
+
+[ -z $valid_left_context ] &&  valid_left_context=$left_context;
+[ -z $valid_right_context ] &&  valid_right_context=$right_context;
+
+[ -z $priors_left_context ] &&  priors_left_context=$left_context;
+[ -z $priors_right_context ] &&  priors_right_context=$right_context;
+
+left_context=$[left_context+frame_subsampling_factor/2]
+right_context=$[right_context+frame_subsampling_factor/2]
+
+egs_opts="--left-context=$left_context --right-context=$right_context --num-frames=$frames_per_eg --num-frames-overlap=$frames_overlap_per_eg --frame-subsampling-factor=$frame_subsampling_factor --compress=$compress $splitter_opts"
+
+valid_left_context=$[valid_left_context+frame_subsampling_factor/2]
+valid_right_context=$[valid_right_context+frame_subsampling_factor/2]
+
+# don't do the overlap thing for the validation data.
+valid_egs_opts="--left-context=$valid_left_context --right-context=$valid_right_context --num-frames=$frames_per_eg --frame-subsampling-factor=$frame_subsampling_factor --compress=$compress $splitter_opts"
+
+priors_left_context=$[priors_left_context+frame_subsampling_factor/2]
+priors_right_context=$[priors_right_context+frame_subsampling_factor/2]
+
+# don't do the overlap thing for the priors computation data.
+priors_egs_opts="--left-context=$priors_left_context --right-context=$priors_right_context --num-frames=1 --compress=$compress"
+
+supervision_all_opts="--frame-subsampling-factor=$frame_subsampling_factor"
+
+echo $left_context > $dir/info/left_context
+echo $right_context > $dir/info/right_context
+
+echo $priors_left_context > $dir/info/priors_left_context
+echo $priors_right_context > $dir/info/priors_right_context
+
+echo $frame_subsampling_factor > $dir/info/frame_subsampling_factor
+
+(
+
+if $adjust_priors && [ $stage -le 10 ]; then
+  
+if [ ! -f $dir/ali.scp ]; then
+  nj_ali=$(cat $alidir/num_jobs)
+  all_ids=$(seq -s, $nj_ali)
+  $cmd $dir/log/copy_alignments.log \
+    copy-int-vector "ark:gunzip -c $alidir/ali.{$all_ids}.gz|" \
+    ark,scp:$dir/ali.ark,$dir/ali.scp || exit 1;
+fi
+
+priors_egs_list=
+for y in `seq $num_archives_priors`; do
+  utils/create_data_link.pl $dir/priors_egs.$y.ark
+  priors_egs_list="$priors_egs_list ark:$dir/priors_egs.$y.ark"
+done
+
+echo "$0: dumping egs for prior adjustment in the background."
+
+num_pdfs=`am-info $alidir/final.mdl | grep pdfs | awk '{print $NF}' 2>/dev/null` || exit 1
+
+$cmd $dir/log/create_priors_subset.log \
+  nnet3-get-egs --num-pdfs=$num_pdfs $priors_ivector_opt $priors_egs_opts "$priors_feats" \
+  "$prior_ali_rspecifier ali-to-post ark:- ark:- |" \
+  ark:- \| nnet3-copy-egs ark:- $priors_egs_list || \
+  { touch $dir/.error; echo "Error in creating priors subset. See $dir/log/create_priors_subset.log"; exit 1; }
+
+sleep 3;
+
+echo $num_archives_priors >$dir/info/num_archives_priors
+
+else
+
```

Comment:
this whole section has odd spacing and intentation.  Please indent blocks inside ( ) and inside if/else, and get rid of the extra spaces around else/if.